sql/src/main/java/io/crate/analyze/TableAnalyzer.java

```diff
@@ -38,15 +38,14 @@
                                               List<Assignment> partitionProperties,
                                               DocTableInfo tableInfo) {
         if (partitionProperties.isEmpty()) {
-            return Arrays.asList(tableInfo.concreteIndices());
+            return Arrays.asList(tableInfo.concreteOpenIndices());
```

Comment:
this change also affects optimize, which would actually work on closed partitions. Not sure what kind of behaviour we want there?

Reply:
Shouldn't we throw an error when all partitions as closed if we try to refresh/optimize a table?

Reply:
The refresh and optimize return the number of tables/partitions refreshed/optimized but maybe would be more insightful for the user to get an error if all partitions are closed as it happens on a non-parted table when it's closed.

Reply:
We generally have the behaviour to ignore unavailable partitions - also for selects, etc. 
I think we should keep this consistent. Good behaviour is predictable and doesn't surprise users. if we were to fail if all partitions are closed, but not if some remain open this could be surprising for users.
And the use case of refreshing the whole table except all closed seems very reasonable, so generally failing if any partition is closed doesn't look appealing to me.

Reply:
My argument is that refresh/optimize as write operations and not select, and the error I feel is correct to be thrown when all partitions are closed, which (to me) feels consistent to a non-parted closed table.

I'm not insisting on this behavior though, as what you said it also feels not "surprising" behavior for the users, so...

Reply:
I tend to agree with @mfussenegger that we should keep the consistent behaviour of ignoring closed partitions. I think a partitioned table with all partitions closed is/behaves still different than a regular closed table.

Reply:
I would definitely keep the same behaviour for `optimize`. I would add a note in docs though to describe it.